src/rebootstrap-raft/main.go

```diff
@@ -123,6 +134,52 @@ func (c *rebootstrapCommand) Run(ctx *cmd.Context) error {
 	return errors.Trace(c.bootstrapRaft(raftServers))
 }
 
+func (c *rebootstrapCommand) getStatePassword(machineID int) (string, error) {
+	agentConfigPath := fmt.Sprintf("/var/lib/juju/agents/machine-%d/agent.conf", machineID)
+	file, err := os.Open(agentConfigPath)
+	if err != nil {
+		return "", errors.Annotatef(err, "opening agent config file %q", agentConfigPath)
+	}
+	defer file.Close()
+
+	var config map[string]interface{}
+	decoder := yaml.NewDecoder(file)
+	if err := decoder.Decode(&config); err != nil {
+		return "", errors.Annotatef(err, "decoding yaml from %q", agentConfigPath)
+	}
+
+	password, ok := config["statepassword"].(string)
+	if !ok || password == "" {
+		return "", errors.Errorf("statepassword not found in %q", agentConfigPath)
+	}
+
+	return password, nil
+}
+
+func (c *rebootstrapCommand) getMachineIDs() (int, error) {
+	agentDirectory := "/var/lib/juju/agents"
```

Comment:
Better to make this a const & we could extract the agent.conf parsing logic from juju/juju / import it to parse and retrieve the machine id rather than do it manually.

Reply:
Cool, that would be much cleaner. I'll work on that.